app/src/main/kotlin/de/markusfisch/android/binaryeye/activity/CameraActivity.kt

```diff
@@ -459,7 +461,18 @@ class CameraActivity : AppCompatActivity() {
 							frameOrientation
 						)?.let { result ->
 							if (result.text != ignoreNext) {
-								postResult(result)
+								if (result.barcodeFormat == BarcodeFormat.PDF_417 && result.text != null) {
+									val iso88591Text = result.text.toByteArray(Charset.forName("ISO-8859-1"))
+									val utf8Text = String(iso88591Text, Charset.forName("UTF-8"))
```

Comment:
From what I have understanding about PDF-417 format - it encodes text always using ISO 8859-1.
Maybe BinaryEye should give user an option to convert to different encoding instead of forcing UTF-8 or use some auto-detection algorithm.